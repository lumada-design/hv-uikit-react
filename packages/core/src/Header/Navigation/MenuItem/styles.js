import "focus-within-polyfill";
import focusStyles from "../../../Focus/styles";

const { focused, focusDisabled, externalReference, falseFocus } = focusStyles;

const styles = (theme) => {
  const hoverColor = theme.hv.palette.atmosphere.atmo3;

  return {
    root: {
      display: "inline",
      "&:hover": {
        "& > [role='button']": {
          backgroundColor: hoverColor,
        },
        "&:focus-within": {
          "& > [role='button']": {
            backgroundColor: hoverColor,
          },
        },
      },
      // IE fallback code (using focus-within-polyfill)
      "&.focus-within": {
        "& > [role='button']": {
          backgroundColor: hoverColor,
        },
      },
    },
    selectedItem: {
      borderTop: `2px solid ${theme.hv.palette.accent.acce3}`,
      paddingTop: "2px",
    },
    notSelectedItem: {
      marginTop: "4px",
    },
    button: {
      border: "none",
      cursor: "pointer",
      padding: `8px ${theme.hv.spacing.sm}px`,
      "&:active": {
        outline: "none",
      },
      "&:focus": {
        backgroundColor: hoverColor,
      },
    },
    contentFocusDisabled: {
      ...focusDisabled,
    },
    contentFocused: {
      ...focused,
    },
    externalReference: {
      ...externalReference,
    },
    falseFocus: {
      ...falseFocus,
      top: "4px",
    },
  };
};

export default styles;
