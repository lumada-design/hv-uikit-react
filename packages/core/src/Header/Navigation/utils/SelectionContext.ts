import { createContext } from "react";

export const SelectionContext = createContext<string[]>([""]);
