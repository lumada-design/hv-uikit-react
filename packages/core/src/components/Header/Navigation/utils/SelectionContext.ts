import { createContext } from "react";

const SelectionContext = createContext<string[]>([]);

export default SelectionContext;
