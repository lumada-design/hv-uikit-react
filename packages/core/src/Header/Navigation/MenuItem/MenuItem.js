import React, { useContext } from "react";
import clsx from "clsx";
import PropTypes from "prop-types";
import { withStyles } from "@mui/styles";
import { KeyboardCodes, isKeypress } from "../../../utils/KeyboardUtils";
import HvTypography from "../../../Typography";
import SelectionContext from "../utils/SelectionContext";
import { FocusContext } from "../utils/FocusContext";
import ConditionalWrapper from "../../../utils/ConditionalWrapper";
import MenuBar from "../MenuBar";
import styles from "./styles";
import isBrowser from "../../../utils/browser";

const MenuItem = ({ classes, id, item, type, onClick }) => {
  const selectionPath = useContext(SelectionContext);
  const { dispatch } = useContext(FocusContext);

  const { data } = item;
  const isMenu = type === "menu";
  const isSelected = selectionPath[isMenu ? 1 : 0] === item.id;
  const hasSubLevel = data && data.length;

  const actionHandler = (event) => {
    if (
      event.type === "click" ||
      isKeypress(event, KeyboardCodes.Enter) ||
      isKeypress(event, KeyboardCodes.SpaceBar)
    ) {
      if (event.type === "click") {
        event.currentTarget.blur();
      }

      onClick?.(event, item);
    }
  };

  const handleFocus = (event) => {
    dispatch({ type: "setItemFocused", itemFocused: event.currentTarget });
  };

  const focusWrapper = (childrenToWrap) => (
    <div className={classes.externalReference}>{childrenToWrap}</div>
  );
  const isIe = isBrowser(["ie", "edge"]);
  const label = (
    <HvTypography variant={isSelected ? "selectedNavText" : "normalText"}>
      {item.label}
    </HvTypography>
  );
  const itemProps = {
    onClick: actionHandler,
    onKeyDown: actionHandler,
    tabIndex: 0,
    onFocus: handleFocus,
  };
  return (
    <li
      id={id}
      key={item.label}
      className={clsx(classes.root, classes[`${type}Item`], {
        [classes.selectedItem]: !isMenu && isSelected,
        [classes.notSelectedItem]: !isMenu && !isSelected,
      })}
    >
      <ConditionalWrapper condition={isIe} wrapper={focusWrapper}>
        {item?.href ? (
          <a
            className={clsx(classes.button, classes.link)}
            href={item?.href}
            target={item?.target}
            {...itemProps}
          >
            {label}
          </a>
        ) : (
          <div className={clsx(classes.button)} role="button" {...itemProps}>
            {label}
          </div>
        )}
      </ConditionalWrapper>
      {hasSubLevel && <MenuBar data={data} onClick={onClick} type="menu" />}
    </li>
  );
};

MenuItem.propTypes = {
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Style applied to the first element in the hierarchy.
     */
    root: PropTypes.string,
    /**
     * Style applied to the li element when it is selected.
     */
    selectedItem: PropTypes.string,
    /**
     * Style applied to the li element when it isn't selected.
     */
    notSelectedItem: PropTypes.string,
    /**
     * Style applied to each item button.
     */
    button: PropTypes.string,
    /**
     * Style applied to each item button when it is behaving as a.
     */
    link: PropTypes.string,
    /**
     * Style applied to the reference element used for ie focus.
     */
    externalReference: PropTypes.string,
    /**
     * Style applied to simulated a focus in ie.
     */
    falseFocus: PropTypes.string,
  }).isRequired,
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * An object containing the data for each menu item.
   *
   * id - the id to be applied to the root element.
   * label - the label to be rendered on the menu item.
   */
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    href: PropTypes.string,
    target: PropTypes.string,
    data: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
      })
    ),
  }).isRequired,
  /**
   * Callback triggered when item is clicked.
   */
  onClick: PropTypes.func,
  /**
   * The type of menu.
   */
  type: PropTypes.oneOf(["menubar", "menu"]).isRequired,
};

export default withStyles(styles, { name: "HvHeaderMenuItem" })(MenuItem);
