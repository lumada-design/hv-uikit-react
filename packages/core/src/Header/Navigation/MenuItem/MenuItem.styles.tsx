import type { CSSInterpolation } from "@emotion/serialize";
import { createClasses } from "@hitachivantara/uikit-react-utils";
import { theme } from "@hitachivantara/uikit-styles";

import { outlineStyles } from "../../../utils/focusUtils";

const item: CSSInterpolation = {
  border: "none",
  cursor: "pointer",
  padding: `${theme.space.xs} ${theme.space.sm}`,
  height: "100%",
  display: "flex",
  alignItems: "center",
  "&:active": {
    outline: "none",
  },
  "&:focus": {
    outline: "none",
  },
  "&:focus-visible": {
    ...outlineStyles,
  },
  "& span": {
    display: "inline-flex",
    color: "inherit",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "space-between",
  },
  "& span::after": {
    content: "attr(data-text)",
    height: 0,
    visibility: "hidden",
    overflow: "hidden",
    userSelect: "none",
    pointerEvents: "none",
    fontWeight: theme.fontWeights.semibold,

    "@media speech": {
      display: "none",
    },
  },
};

export const { staticClasses, useClasses } = createClasses("HvHeaderMenuItem", {
  root: {
    display: "inline",
    height: "100%",
    borderBottom: "4px solid transparent",

    ":hover, :focus-within": {
      backgroundColor: theme.colors.bgHover,
    },
  },
  menu: {},
  menubar: {},
  selected: {
    color: theme.colors.text,
    borderColor: "currentcolor",
  },
  item,
  link: {
    textDecoration: "none",
  },
  button: {
    color: "inherit",
  },
});
