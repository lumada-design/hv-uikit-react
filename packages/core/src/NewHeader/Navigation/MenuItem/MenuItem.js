/*
 * Copyright 2020 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useContext } from "react";
import classNames from "classnames";
import PropTypes from "prop-types";
import withStyles from "@material-ui/core/styles/withStyles";
import { KeyboardCodes, isKeypress } from "@hv/uikit-common-utils/dist";
import HvTypography from "../../../Typography";
import useUniqueId from "../../../useUniqueId";
import SelectionContext from "../utils/SelectionContext";
import { FocusContext } from "../utils/FocusContext";
import MenuBar from "../MenuBar";

import styles from "./styles";

const MenuItem = ({ classes, id, item, type, onClick }) => {
  const uniqueId = useUniqueId(id, "hv-menuItem-");
  const selectionPath = useContext(SelectionContext);
  const { dispatch } = useContext(FocusContext);

  const { data } = item;
  const isMenu = type === "menu";
  const isSelected = selectionPath[isMenu ? 1 : 0] === item.id;
  const hasSubLevel = data && data.length;

  const actionHandler = event => {
    if (
      event.type === "click" ||
      isKeypress(event, KeyboardCodes.Enter) ||
      isKeypress(event, KeyboardCodes.SpaceBar)
    ) {
      if (event.type === "click") {
        // hide focus outline when using mouse
        event.currentTarget.blur();
      }

      if (onClick) {
        onClick(event, item);
      }
    }
  };

  const handleFocus = event => {
    dispatch({ type: "setItemFocused", itemFocused: event.currentTarget });
  };

  return (
    <li
      id={uniqueId}
      key={item.label}
      role="none"
      className={classNames(
        classes.root,
        classes[`${type}Item`],
        isSelected ? classes.selectedItem : undefined
      )}
    >
      <div
        role="button"
        className={classes.button}
        onClick={actionHandler}
        onKeyDown={actionHandler}
        tabIndex={0}
        onFocus={handleFocus}
      >
        <HvTypography variant={isSelected ? "selectedNavText" : "normalText"}>
          {item.label}
        </HvTypography>
      </div>
      {hasSubLevel && <MenuBar data={data} onClick={onClick} type="menu" />}
    </li>
  );
};

MenuItem.propTypes = {
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    root: PropTypes.string,
    menubarItem: PropTypes.string,
    selectedItem: PropTypes.string,
    menuItem: PropTypes.string,
    button: PropTypes.string
  }).isRequired,
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * An object containing the data for each menu item.
   *
   * id - the id to be applied to the root element.
   * label - the label to be rendered on the menu item.
   */
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired
  }).isRequired,
  /**
   * Callback triggered when item is clicked.
   */
  onClick: PropTypes.func,
  /**
   * The type of menu.
   */
  type: PropTypes.oneOf(["menubar", "menu"]).isRequired
};

MenuItem.defaultProps = {
  id: undefined,
  onClick: () => {}
};

export default withStyles(styles, { name: "HvNewHeaderMenuItem" })(MenuItem);
