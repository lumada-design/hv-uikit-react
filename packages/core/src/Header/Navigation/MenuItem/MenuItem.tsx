import React, { MouseEvent, useContext } from "react";

import { HvTypography } from "../../../Typography";
import { HvBaseProps } from "../../../types/generic";
import { isKey } from "../../../utils/keyboardUtils";
import { ExtractNames } from "../../../utils/classes";

import { useDefaultProps } from "../../../hooks/useDefaultProps";

import { FocusContext } from "../utils/FocusContext";
import { SelectionContext } from "../utils/SelectionContext";
import { useClasses, staticClasses } from "./MenuItem.styles";
import { HvHeaderNavigationItemProp } from "../useSelectionPath";
import { Bar } from "../MenuBar/Bar";

export { staticClasses as headerMenuItemClasses };

export type HvHeaderMenuItemClasses = ExtractNames<typeof useClasses>;

export interface HvHeaderMenuItemProps
  extends HvBaseProps<HTMLDivElement, "onClick"> {
  item: HvHeaderNavigationItemProp;
  type?: string;
  onClick?: (event: MouseEvent, selection: HvHeaderNavigationItemProp) => void;
  levels: number;
  currentLevel: number;
  classes?: HvHeaderMenuItemClasses;
}

// Traverse the tree of items and return the first href it finds
const traverseItem = (
  node: HvHeaderNavigationItemProp
): { href?: string; target?: string } => {
  let href: string | undefined;
  let target: string | undefined;

  if (node?.href) {
    href = node?.href;
    target = node?.target;
  } else if (node?.data != null && node?.data?.length > 0) {
    let i = 0;
    while (href == null && i < node.data.length) {
      const childNavItem = traverseItem(node?.data[i]);
      if (childNavItem?.href != null) {
        href = childNavItem?.href;
        target = childNavItem?.target;
        break;
      }

      i += 1;
    }
  }

  return { href, target };
};

export const HvHeaderMenuItem = (props: HvHeaderMenuItemProps) => {
  const {
    id,
    item,
    type,
    onClick,
    levels,
    currentLevel,
    classes: classesProp,
    className,
  } = useDefaultProps("HvHeaderMenuItem", props);
  const { classes, cx } = useClasses(classesProp);

  const selectionPath = useContext(SelectionContext);

  const { dispatch } = useContext(FocusContext);

  const { data } = item;

  const hasSubLevel = data && data.length;

  const isMenu = type === "menu";

  const isSelected = selectionPath?.[isMenu ? 1 : 0] === item.id;

  // true: if the item is part of the selection path but is not the current page the user is seeing, i.e has more sub levels
  // page: used when the selected item is actually the current page the user is seeing
  const isCurrent = isSelected ? (hasSubLevel ? true : "page") : undefined;

  const actionHandler = (event: any) => {
    if (
      event.type === "click" ||
      isKey(event, "Enter") ||
      isKey(event, "Space")
    ) {
      if (event.type === "click") {
        event.currentTarget.blur();
      }

      onClick?.(event, item);
    }
  };

  const handleFocus = (event: React.FocusEvent) => {
    dispatch?.({ type: "setItemFocused", itemFocused: event.currentTarget });
  };

  const itemProps = {
    onClick: actionHandler,
    onKeyDown: actionHandler,
    onFocus: handleFocus,
  };

  const label = (
    <HvTypography
      component="span"
      variant={isSelected ? "label" : "body"}
      data-text={item.label}
    >
      {item.label}
    </HvTypography>
  );

  let itemHref = item?.href;
  let itemTarget = item?.target;

  // apps should configure the href even on parent items without content
  // so the fallback logic is theirs, but if not we'll do our best to find a link
  if (item?.href == null) {
    const { href, target } = traverseItem(item);
    itemHref = href;
    itemTarget = target;
  }

  return (
    <li
      id={id}
      key={item.label}
      className={cx(
        classes.root,
        {
          [classes.menu]: isMenu,
          [classes.menubar]: !isMenu,
          [classes.selected]: !!isSelected,
        },
        className
      )}
    >
      {itemHref ? (
        <a
          className={classes.link}
          href={itemHref}
          target={itemTarget}
          {...itemProps}
          aria-current={isCurrent}
          // Fix for a possible bug in playwright where, even though hidden from screen readers and not appearing in the accessibility
          // tree, due to the styling duplication (span::after - using 'data-text'), locators have to double the name.
          aria-label={item.label}
        >
          {label}
        </a>
      ) : (
        // keeping this code path for backwards compatibility, but
        // shouldn't really be used as it's not accessible
        <div
          className={classes.button}
          role="button"
          {...itemProps}
          tabIndex={0}
          aria-current={isCurrent}
        >
          {label}
        </div>
      )}
      {/* Limits levels to no more than 2. More than that is not expected and not in DS. */}
      {hasSubLevel && currentLevel < levels && currentLevel < 2 && (
        <Bar data={data} type="menu">
          {data.map((itm: HvHeaderNavigationItemProp) => (
            <HvHeaderMenuItem
              key={itm.id}
              item={itm}
              type="menu"
              onClick={onClick}
              levels={levels}
              currentLevel={currentLevel + 1}
            />
          ))}
        </Bar>
      )}
    </li>
  );
};
