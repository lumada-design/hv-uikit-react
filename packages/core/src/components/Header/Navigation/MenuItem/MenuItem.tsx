import { Typography } from "components/Typography";
import { useContext } from "react";
import { isKeypress, KeyboardCodes } from "utils/KeyboardUtils";
import { MenuBar } from "..";
import { NavigationItemProp } from "../Navigation";
import { FocusContext } from "../utils/FocusContext";
import SelectionContext from "../utils/SelectionContext";
import { MenuItemLabel, MenuItemLi, MenuItemLink } from "./MenuItem.styles";

export interface MenuItemProps extends BaseProps<"div", { onClick }> {
  item: NavigationItemProp;
  type?: string;
  onClick?: (event: MouseEvent, selection: NavigationItemProp) => void;
}

export const MenuItem = ({ id, item, type, onClick }: MenuItemProps) => {
  const selectionPath = useContext(SelectionContext);
  // @ts-ignore
  const { dispatch } = useContext(FocusContext);

  const { data } = item;
  const hasSubLevel = data && data.length;
  const isMenu = type === "menu";
  const isSelected =
    (selectionPath && selectionPath[isMenu ? 1 : 0] === item.id) || false;

  const actionHandler = (event) => {
    if (
      event.type === "click" ||
      isKeypress(event, KeyboardCodes.Enter) ||
      isKeypress(event, KeyboardCodes.SpaceBar)
    ) {
      if (event.type === "click") {
        event.currentTarget.blur();
      }
      onClick?.(event, item);
    }
  };

  const handleFocus = (event) => {
    dispatch({ type: "setItemFocused", itemFocused: event.currentTarget });
  };

  const itemProps = {
    onClick: actionHandler,
    onKeyDown: actionHandler,
    tabIndex: 0,
    onFocus: handleFocus,
  };

  const label = (
    <Typography variant={isSelected ? "label" : "body"}>
      {item.label}
    </Typography>
  );

  return (
    <MenuItemLi
      id={id}
      key={item.label}
      selected={!isMenu && isSelected ? "selectedItem" : "notSelectedItem"}
    >
      {item?.href ? (
        <MenuItemLink href={item?.href} target={item?.target} {...itemProps}>
          {label}
        </MenuItemLink>
      ) : (
        <MenuItemLabel role="button" {...itemProps} isSelected={isSelected}>
          {label}
        </MenuItemLabel>
      )}
      {hasSubLevel && <MenuBar data={data} onClick={onClick} type="menu" />}
    </MenuItemLi>
  );
};
