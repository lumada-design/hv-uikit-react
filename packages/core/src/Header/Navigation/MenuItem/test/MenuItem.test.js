import React from "react";
import { mount } from "enzyme";

import HvProvider from "../../../../Provider";
import SelectionContext from "../../utils/SelectionContext";
import MenuItem from "../index";

describe("MenuItem withStyles", () => {
  let wrapper;

  it("should be able to render", () => {
    const contextValue = ["someId"];

    wrapper = mount(
      <HvProvider>
        <SelectionContext.Provider value={contextValue}>
          <MenuItem item={{ id: "someId ", label: "someLabel" }} type="menubar" />
        </SelectionContext.Provider>
      </HvProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
