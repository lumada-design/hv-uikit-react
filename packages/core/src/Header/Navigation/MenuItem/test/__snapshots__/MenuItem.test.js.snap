// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem withStyles should be able to render 1`] = `
<WithStyles(MenuItem)
  item={
    Object {
      "id": "someId ",
      "label": "someLabel",
    }
  }
  type="menubar"
>
  <MenuItem
    classes={
      Object {
        "button": "HvHeaderMenuItem-button",
        "contentFocusDisabled": "HvHeaderMenuItem-contentFocusDisabled",
        "contentFocused": "HvHeaderMenuItem-contentFocused",
        "externalReference": "HvHeaderMenuItem-externalReference",
        "falseFocus": "HvHeaderMenuItem-falseFocus",
        "notSelectedItem": "HvHeaderMenuItem-notSelectedItem",
        "root": "HvHeaderMenuItem-root",
        "selectedItem": "HvHeaderMenuItem-selectedItem",
      }
    }
    item={
      Object {
        "id": "someId ",
        "label": "someLabel",
      }
    }
    type="menubar"
  >
    <li
      className="HvHeaderMenuItem-root HvHeaderMenuItem-notSelectedItem"
      key="someLabel"
    >
      <ConditionalWrapper
        condition={null}
        wrapper={[Function]}
      >
        <div
          className="HvHeaderMenuItem-button HvHeaderMenuItem-contentFocusDisabled"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <WithStyles(ForwardRef)
            variant="normalText"
          >
            <ForwardRef
              classes={
                Object {
                  "3xlTitle": "HvTypography-3xlTitle",
                  "4xlTitle": "HvTypography-4xlTitle",
                  "5xlTitle": "HvTypography-5xlTitle",
                  "disabledButtonText": "HvTypography-disabledButtonText",
                  "highlightText": "HvTypography-highlightText",
                  "infoText": "HvTypography-infoText",
                  "inlineLink": "HvTypography-inlineLink",
                  "lTitle": "HvTypography-lTitle",
                  "labelText": "HvTypography-labelText",
                  "mTitle": "HvTypography-mTitle",
                  "normalText": "HvTypography-normalText",
                  "placeholderText": "HvTypography-placeholderText",
                  "root": "HvTypography-root",
                  "sLink": "HvTypography-sLink",
                  "sText": "HvTypography-sText",
                  "sTitle": "HvTypography-sTitle",
                  "selectedNavText": "HvTypography-selectedNavText",
                  "selectedText": "HvTypography-selectedText",
                  "vizText": "HvTypography-vizText",
                  "xlTitle": "HvTypography-xlTitle",
                  "xsTitle": "HvTypography-xsTitle",
                  "xxlTitle": "HvTypography-xxlTitle",
                  "xxsTitle": "HvTypography-xxsTitle",
                }
              }
              variant="normalText"
            >
              <p
                className="HvTypography-root HvTypography-normalText"
              >
                someLabel
              </p>
            </ForwardRef>
          </WithStyles(ForwardRef)>
        </div>
      </ConditionalWrapper>
    </li>
  </MenuItem>
</WithStyles(MenuItem)>
`;
